extensions/amp-consent/0.1/consent-ui.js

```diff
@@ -270,14 +270,34 @@ export class ConsentUI {
    */
   createPromptIframeFromSrc_(promptUISrc) {
     const iframe = this.parent_.ownerDocument.createElement('iframe');
+    let sandbox = 'allow-scripts';
     iframe.src = assertHttpsUrl(promptUISrc, this.parent_);
-    iframe.setAttribute('sandbox', 'allow-scripts');
+    const allowSameOrigin = this.allowSameOrigin_(
+        iframe.src, window.location.href);
+    if (allowSameOrigin) {
+      sandbox = 'allow-scripts allow-same-origin';
+    }
+    iframe.setAttribute('sandbox', sandbox);
     const {classList} = iframe;
     classList.add(consentUiClasses.fill);
     // Append iframe lazily to save resources.
     return iframe;
   }
 
+  /**
+   * Determines if allow-same-origin should be enabled for the prompt iframe
+   * @param {string} src
+   * @param {string} containerSrc
+   * @return {boolean}
+   */
+  allowSameOrigin_(src, containerSrc) {
+    const urlService = Services.urlForDoc(this.parent_);
+    const srcUrl = urlService.parse(src);
+    const containerUrl = urlService.parse(containerSrc);
+
+    return srcUrl.origin != containerUrl.origin && srcUrl.protocol != 'data:';
```

Comment:
nit: we don't need to check for `srcUrl.protocoal == 'data:'`, because there's `assertHttpsUrl` already.